authorization/v2/parser.go

```diff
@@ -0,0 +1,89 @@
+package v2
+
+import (
+	"strings"
+)
+
+// NewURIFromString parses the string representation of a URI into the URI struct.
+// NewURIFromString expects the string to be of the following form, otherwise ErrInvalidURI is returned
+// hs:<service_name>:<subsystem>:<version>:<category>:<resource_name>?<allowed_arguments>#<permission_metadata>
+func NewURIFromString(source string) (UniformResourceIdentifier, error) {
+	var (
+		arguments map[string]string = nil
+		metadata  map[string]string = nil
+	)
```

Comment:
```
var (
		arguments map[string]string
		metadata  map[string]string
)
```
Would work fine too since `nil` is the zero value for maps.

Reply:
Oh cool, I didn't know that!